internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,447 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package delegatingresolver_test
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/internal/pretty"
+	"google.golang.org/grpc/internal/proxyattributes"
+	"google.golang.org/grpc/internal/resolver/delegatingresolver"
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+
+	_ "google.golang.org/grpc/resolver/dns" // To register dns resolver.
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+const (
+	targetTestAddr          = "test.com"
+	resolvedTargetTestAddr1 = "1.2.3.4:8080"
+	resolvedTargetTestAddr2 = "1.2.3.5:8080"
+	envProxyAddr            = "proxytest.com"
+	resolvedProxyTestAddr1  = "2.3.4.5:7687"
+	resolvedProxyTestAddr2  = "2.3.4.6:7687"
```

Comment:
nit: Please move these constants into individual tests that use them. This provides 2 benefits:
1. Let redirection when reading the tests.
2. The global namespace doesn't get polluted as the number of test functions increase..

Reply:
It's ok for test code to be repetitive. See: https://engdoc.corp.google.com/eng/doc/tott/episodes/643.md?cl=head